### PR TITLE
add model and test data for fixing dtype bug in nary eltwise layers

### PR DESCRIPTION
**Pull request by fengyuentau**
Merge with https://github.com/opencv/opencv/pull/24386

Also noted that random seed is added in this pr, so previous data is regenerated to keep consistancy.

### CODE REVIEW

File: testdata/dnn/onnx/data/output_gather_shared_indices.npy
Comment (dkurt) [inline testdata/dnn/onnx/data/output_gather_shared_indices.npy:1]: This file and `testdata/dnn/onnx/data/input_gather_shared_indices.npy` are empty. Is that required to for a test?

Comment (fengyuentau) [reply testdata/dnn/onnx/data/output_gather_shared_indices.npy:1]: > Also noted that random seed is added in this pr, so previous data is regenerated to keep consistancy.

I added the random seed, and previous data needs to be regenerated.

**Comment by asmorkalov** [comment]
@dkurt @fengyuentau Is it ready for merge?

File: testdata/dnn/onnx/data/output_gather_shared_indices.npy
Comment (fengyuentau) [reply testdata/dnn/onnx/data/output_gather_shared_indices.npy:1]: Note that they are not empty though, they are `+0 Bytes).

**Comment by fengyuentau** [comment]
> @dkurt @fengyuentau Is it ready for merge?

I think yes, all tests are green.

File: testdata/dnn/onnx/data/output_gather_shared_indices.npy
Comment (dkurt) [reply testdata/dnn/onnx/data/output_gather_shared_indices.npy:1]: Oh, I see, excuse me. that's a problem of GitHub,
```
$ du -hs -b testdata/dnn/onnx/data/*gather_shared*
224     testdata/dnn/onnx/data/input_gather_shared_indices.npy
176     testdata/dnn/onnx/data/output_gather_shared_indices.npy
```

**Review by dkurt — APPROVED** [review]

**Pull request merged**